coarse/api/aspect_purpose.py

```
@@ -0,0 +1,19 @@
+from coarse.models.aspect_purpose import AspectPurposeOutput
+import json
+from pprint import pprint
+router = APIRouter()
+
+@router.post("/aspect_purpose", response_model=AspectPurposeOutput)
+async def aspect_purpose_route(paper_obj: SectionMap):
+    try:
+        result = await aspect_purpose(paper_obj)
+        return result
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"An error occurred while processing the PDF: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)